src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/Maintainability/UseCrossPlatformIntrinsicsAnalyzer.cs

```diff
@@ -0,0 +1,353 @@
+                AddBinaryOperatorMethods(methodSymbolSets, "And", x86SseTypeSymbol, RuleKind.op_BitwiseAnd);
+                AddBinaryOperatorMethods(methodSymbolSets, "Divide", x86SseTypeSymbol, RuleKind.op_Division);
+                AddBinaryOperatorMethods(methodSymbolSets, "Multiply", x86SseTypeSymbol, RuleKind.op_Multiply);
+                AddBinaryOperatorMethods(methodSymbolSets, "Or", x86SseTypeSymbol, RuleKind.op_BitwiseOr);
+                AddBinaryOperatorMethods(methodSymbolSets, "Subtract", x86SseTypeSymbol, RuleKind.op_Subtraction);
+                AddBinaryOperatorMethods(methodSymbolSets, "Xor", x86SseTypeSymbol, RuleKind.op_ExclusiveOr);
+            }
+
+            if (compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeIntrinsicsX86Sse2, out var x86Sse2TypeSymbol))
+            {
+                AddBinaryOperatorMethods(methodSymbolSets, "Add", x86Sse2TypeSymbol, RuleKind.op_Addition);
+                AddBinaryOperatorMethods(methodSymbolSets, "And", x86Sse2TypeSymbol, RuleKind.op_BitwiseAnd);
+                AddBinaryOperatorMethods(methodSymbolSets, "Divide", x86Sse2TypeSymbol, RuleKind.op_Division);
+                AddBinaryOperatorMethods(methodSymbolSets, "Multiply", x86Sse2TypeSymbol, RuleKind.op_Multiply, [SpecialType.System_Double]);
+                AddBinaryOperatorMethods(methodSymbolSets, "MultiplyLow", x86Sse2TypeSymbol, RuleKind.op_Multiply);
+                AddBinaryOperatorMethods(methodSymbolSets, "Or", x86Sse2TypeSymbol, RuleKind.op_BitwiseOr);
+                AddBinaryOperatorMethods(methodSymbolSets, "Subtract", x86Sse2TypeSymbol, RuleKind.op_Subtraction);
+                AddBinaryOperatorMethods(methodSymbolSets, "Xor", x86Sse2TypeSymbol, RuleKind.op_ExclusiveOr);
+
+                AddShiftOperatorMethods(methodSymbolSets, "ShiftLeftLogical", x86Sse2TypeSymbol, RuleKind.op_LeftShift);
+                AddShiftOperatorMethods(methodSymbolSets, "ShiftRightArithmetic", x86Sse2TypeSymbol, RuleKind.op_RightShift);
+                AddShiftOperatorMethods(methodSymbolSets, "ShiftRightLogical", x86Sse2TypeSymbol, RuleKind.op_UnsignedRightShift);
+            }
+
+            if (compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemRuntimeIntrinsicsX86Sse41, out var x86Sse41TypeSymbol))
+            {
+                AddBinaryOperatorMethods(methodSymbolSets, "MultiplyLow", x86Sse41TypeSymbol, RuleKind.op_Multiply);
+            }
+
+            if (methodSymbolSets.Any((methodSymbols) => methodSymbols.Any()))
+            {
+                context.RegisterOperationAction((context) => AnalyzeInvocation(context, methodSymbolSets), OperationKind.Invocation);
+            }
+
+            static void AddBinaryOperatorMethods(ImmutableArray<HashSet<IMethodSymbol>> methodSymbolSets, string name, INamedTypeSymbol typeSymbol, RuleKind ruleKind, params SpecialType[] supportedTypes)
```

Comment:
```suggestion
            return;

            static void AddBinaryOperatorMethods(ImmutableArray<HashSet<IMethodSymbol>> methodSymbolSets, string name, INamedTypeSymbol typeSymbol, RuleKind ruleKind, params SpecialType[] supportedTypes)
```

The explicit `return;` makes it clear in the code that there is no more 'logic' following that one has to keep in mind and look for.  Effectively all that follows are local functions.  this helps with mental tracking things.